pkg/chunk/cached_store.go

```diff
@@ -921,6 +926,20 @@ func (store *cachedStore) regMetrics(reg prometheus.Registerer) {
 		func() float64 {
 			return float64(len(store.currentUpload))
 		}))
+	reg.MustRegister(prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Name:        "store_readers",
+		Help:        "number of storage readers",
+		ConstLabels: prometheus.Labels{"storage": "cache"},
+	}, func() float64 {
+		return float64(store.cacheReaders.Load())
```

Comment:
We can know the concurrent IO from `dcState`

Reply:
`concurrency` counter is only updated in `unstableDC` implementation. To track concurrency in other states, I have to move it to `baseDC`?